### PR TITLE
Add type checking to checks and fixed the bugs found by it

### PR DESCRIPTION
**Pull request by TomCusack-Softwire**
Now GitHub actions should fail if TypeScript doesn't compile :)

### CODE REVIEW

File: src/components/Tables/Table.tsx [hunk 1]
```diff
@@ -48,7 +48,8 @@ const dataToRows = (data: Datum[], headers: Headers): Row[] => {
         const row: Row = { rowId: currentIndex };
 
         for (const headerKey of Object.keys(headers)) {
-            row[headerKey] = datum[headerKey] ?? "";
+            const databaseValue = datum[headerKey] ?? "";
+            row[headerKey] = Array.isArray(databaseValue) ? databaseValue.join() : databaseValue;
```
Comment (daniel-conlan-softwire) [inline src/components/Tables/Table.tsx:52]: Should the `.join()` function be replaced with `.join(", ")`? Might look a bit nicer with a space between list elements, but it may not matter too much

**Review by daniel-conlan-softwire — APPROVED** [review]
All looks good

File: src/components/Tables/Table.tsx (hunk 1 above)
Comment (paopaosoftwire) [inline src/components/Tables/Table.tsx]: Is the new section arises from the database changes? Should this be in a separate PR if it does not relate to type checking? Just afraid that this might effect others that are using the Table component.

Comment (TomCusack-Softwire) [reply src/components/Tables/Table.tsx]: It maybe should be in another PR, but it does directly relate to the type checking - without this, the type checker fails!
Previously it didn't check types 'that well' so when the database types changed it allowed it - but there is failing code on main! :eyes:
